starport/interface/cli/starport/cmd/network_chain_show.go

```diff
@@ -17,36 +27,99 @@ func NewNetworkChainShow() *cobra.Command {
 		RunE:  networkChainShowHandler,
 		Args:  cobra.ExactArgs(1),
 	}
+	c.Flags().Bool(genesisFlag, false, "Show exclusively the genesis of the chain")
+	c.Flags().Bool(peersFlag, false, "Show exclusively the peers of the chain")
 	return c
 }
 
 func networkChainShowHandler(cmd *cobra.Command, args []string) error {
+	chainID := args[0]
+
 	nb, err := newNetworkBuilder()
 	if err != nil {
 		return err
 	}
 
-	chainID := args[0]
-
-	c, err := nb.ShowChain(context.Background(), chainID)
+	// Get flags
+	showGenesis, err := cmd.Flags().GetBool(genesisFlag)
 	if err != nil {
 		return err
 	}
-	chainyaml, err := yaml.Marshal(c)
+	showPeers, err := cmd.Flags().GetBool(peersFlag)
 	if err != nil {
 		return err
 	}
+	if showGenesis && showPeers {
+		return fmt.Errorf("%s and %s flags cannot be used together", genesisFlag, peersFlag)
+	}
 
-	info, err := nb.LaunchInformation(context.Background(), chainID)
+	// Fetch launch information
+	info, err := nb.LaunchInformation(cmd.Context(), chainID)
 	if err != nil {
 		return err
 	}
-	infoyaml, err := yaml.Marshal(info)
+
+	switch {
+	case showGenesis:
+		if err := generateAndShowGenesis(cmd, nb, chainID, info); err != nil {
+			return err
+		}
+	case showPeers:
+		// Show the peers in the config.toml format
+		fmt.Printf("persistent_peers = \"%s\"\n", strings.Join(info.Peers, ","))
+	default:
+		// No flag, show the chain information and launch information in yaml format
+		chain, err := nb.ShowChain(cmd.Context(), chainID)
+		if err != nil {
+			return err
+		}
+		chainyaml, err := yaml.Marshal(chain)
+		if err != nil {
+			return err
+		}
+		infoyaml, err := yaml.Marshal(info)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("\nChain:\n---\n%s\n\nLaunch Information:\n---\n%s", string(chainyaml), string(infoyaml))
+	}
+
+	return nil
+}
+
+func generateAndShowGenesis(cmd *cobra.Command, nb *networkbuilder.Builder, chainID string, info spn.LaunchInformation) error {
+	tmpHome, err := os.MkdirTemp("", "")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpHome)
+
+	// Initialize the blockchain
+	blockchain, err := nb.Init(
+		cmd.Context(),
+		chainID,
+		networkbuilder.SourceChainID(),
+		networkbuilder.InitializationHomePath(tmpHome),
+	)
+	if err == context.Canceled {
```

Comment:
```suggestion
	if err != nil { return err }
```

seems like we have a final canceled [check](https://github.com/tendermint/starport/blob/cf58333/starport/interface/cli/starport/main.go#L40)